src/validation-dispatcher/single-field-dispatcher.ts

```diff
@@ -36,23 +24,24 @@ const checkValidationResult = (
 const fireValidation = (
   value,
   values,
-  fieldValidation: FieldValidation
+  fieldValidation: FullFieldValidationAsync
 ): Promise<ValidationResult> => {
-  const validationFn = getValidationFn(fieldValidation);
-  return validationFn(
-    value,
-    values,
-    fieldValidation['customArgs'],
-    fieldValidation['message']
-  ).then(checkValidationResult);
+  return fieldValidation
+    .validator(
+      value,
+      values,
+      fieldValidation['customArgs'],
```

Comment:
Here we can ensure that it exists `fieldValidation.customArgs` and `fieldValidation.message`, isn't it?

Reply:
and if not return undefined? maybe  is what we get by default, altough there could be some case where it could fail?

something like?

```
(fieldVaildation['customArgs']) f? ieldValidation['customArgs'] : undefined
```

Reply:
I think I'm gong to leave it like it is now, if there is something I'm missing or other proposal, please gimme a buzz we can open branch and refactor